engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -1058,33 +1018,97 @@ verifyEscape = do
-  lvl <- getLevel lidV
   K.PointUI x y <- getsSession spointer
   let (px, py) = (x `div` 2, y - K.mapStartY)
       tpos = Point px py
-      t = lvl `at` tpos
   if px >= 0 && py >= 0 && px < rXmax && py < rYmax
-  then alterTileAtPos ts tpos $ "the" <+> TK.tname (okind cotile t)
-  else failWith "never mind"
+  then alterTileAtPos (Just tpos)
+  else alterTileAtPos Nothing
+
+-- * CloseDir 
+
+-- | Close nearby open tile; ask for direction, if there is more than one.
+closeDirHuman :: MonadClientUI m
+              => m (FailOrCmd RequestTimed)
+closeDirHuman = do
+  COps{cotile} <- getsState scops
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  lvl <- getLevel $ blid b
+  let vPts = vicinityUnsafe $ bpos b 
+      openPts = filter (Tile.isClosable cotile . at lvl) vPts
+  case openPts of 
+    []  -> failSer CloseNothing
+    [o] -> closeTileAtPos $ Just o
+    _   -> pickPoint "close" >>= closeTileAtPos 
+   
+-- | Close tile at given position.
+closeTileAtPos :: MonadClientUI m
+               => Maybe Point -> m (FailOrCmd RequestTimed)
+closeTileAtPos mp = case mp of
+  Nothing -> failWith "never mind"
+  Just p -> do
+    COps{cotile} <- getsState scops
+    leader <- getLeaderUI
+    b <- getsState $ getActorBody leader
+    if (p `chessDist` bpos b == 1) then do
+      actorSk <- leaderSkillsClientUI
+      lvl <- getLevel $ blid b
+      let alterSkill = Ability.getSk Ability.SkAlter actorSk
+      if | Tile.isOpenable cotile $ lvl `at` p 
+          -> failSer CloseClosed
+         | alterSkill <= 1
+          -> failSer AlterUnskilled
+         | otherwise 
+          -> do
+             msgAddDone p "close"
+             return $ Right $ ReqAlter p 
+    else failSer CloseDistant 
+
+-- | Adds message with proper names.
+msgAddDone :: MonadClientUI m
+           => Point -> Text
+           -> m ()
+msgAddDone p verb = do
+  COps{cotile} <- getsState scops
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  lvl <- getLevel $ blid b
+  let s = case (T.words . TK.tname $ okind cotile $ lvl `at` p) of
+          [] -> "thing"
+          [x] -> x
+          (_ : xs) -> T.unwords xs
+      dir = compassText $ p `vectorToFrom` bpos b
+  msgAdd MsgDone $ "you" <+> verb <+> "the" <+> s <+> dir
+
+-- | Prompts user to pick a point.
+pickPoint :: MonadClientUI m
+          => Text -> m (Maybe Point)
+pickPoint verb = do
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  UIOptions{uVi, uLeftHand} <- getsSession sUIOptions
+  let keys = K.escKM
+           : K.leftButtonReleaseKM
+           : map (K.KM K.NoModifier) (K.dirAllKey uVi uLeftHand)
+  promptAdd0 $ "Where to" <+> verb <> "? [movement key] [pointer]"
+  slides <- reportToSlideshow [K.escKM]
+  km <- getConfirms ColorFull keys slides
+  case K.key km of
+    K.LeftButtonRelease -> do
+      K.PointUI x y <- getsSession spointer
+      let p = Point (x `div` 2) (y - K.mapStartY)
+      if (p == bpos b) then return Nothing
```

Comment:
Parens not needed. BTW, I wonder how strict this check should be. Another alternative is not to check at all here, yet another is to check if `chessDist p (bpos b) == 1`. Let's see how this is checked again in the code later on...

Reply:
`chessDist p (bpos b) == 1` is checked after we make use of `pickPoint`, at lines 1062 and 959 precisely . It makes sure that player can't pick the point where his character is standing. Otherwise we would get another error message, `CloseClosed` afair.

Reply:
Actually, this is probably perfect, because it only rejects the clearly forbidden action: altering terrain under the player character. Such an action is only possible with S-RMB, and I'm not even sure I will keep that command.

And less nonsensical actions are checked for in other code, where they get varied error messages to help a newbie player. Great.

Reply:
I've checked (standing over a water puddle in crawl scenario 3rd level) and right now it's not even possible to alter terrain under an actor even with S-RMB. So I think it's fair to ignore that possibility.

Reply:
Right, let's keep it as is: here we make sure position under player can't be chosen ("never mind", because it's clearly nonsensical) and later on we reject far position (picked with the mouse), giving a helpful error message about not enough reach to alter at a distance, etc. Perfect.

Reply:
I wondered if it wouldn't be better to let that way to hide player characters, e.g. when one stands _in_ a chest.

Reply:
Heh, that's a nice idea, but it would require some serious brainstorming. E.g., right now, only terrain between viewer and a target position hides that position from LOS, not terrain at either the viewer or the target position. In other words, actors and items on tiles are supposed to touch the borders on a tile and so be see and be visible. Also, it's a pain to mark on the map actors standing inside solid tiles and if we don't mark it, it's a bit useless and misleading (it can always be read by pointing the aiming crosshair at the tile, but we don't want to encourage the player to point all over the map each move).

So to have a chest that hides the player, the easiest way would be to have chest wall tiles all around the chest interior tile. And it can be done already. Perhaps a big wardrobe would be a better fit than a chest.